src/transformers/trainer.py

```diff
@@ -3201,12 +3246,24 @@ def training_step(self, model: nn.Module, inputs: Dict[str, Union[torch.Tensor,
 
                 The dictionary will be unpacked before being fed to the model. Most models expect the targets under the
                 argument `labels`. Check your model's documentation for all accepted arguments.
+            kwargs:
+                Additional key-word arguments to pass along for custom optimizers
 
         Return:
             `torch.Tensor`: The tensor with training loss on this batch.
         """
         model.train()
         inputs = self._prepare_inputs(inputs)
+        _is_lomo = False
+
+        if is_lomo_available():
+            from lomo_optim import AdaLomo, Lomo
+
+            _is_lomo = isinstance(_unwrap_optimizer(self.optimizer), (Lomo, AdaLomo))
```

Comment:
I'd move out the common logic to something like `_is_lomo_optimizer` from here and L1086, which handles importing Lomo and AdaLomo and unwrapping the optimizer